File: algorithms/2_medium/M/02086_minimum_number_of_food_buckets_to_feed_the_hamsters/solution.c
#define HAMSTER 'H'
#define EMPTY '.'

int minimumBuckets(const char *restrict hamsters) {
    const size_t len = strlen(hamsters);
    size_t i = 0;
    size_t cnt = 0;

    while (i < len) {
        if (hamsters[i] == HAMSTER) {
            if (i + 1 < len && hamsters[i + 1] == EMPTY) {
                i += 3;
                cnt += 1;
            } else if (i > 0 && hamsters[i - 1] == EMPTY) {
                cnt += 1;
                i += 1;
            } else {
                return -1;
            }
        } else {
            i += 1;
        }
    }

    return (int)cnt;
}

File: algorithms/2_medium/M/02086_minimum_number_of_food_buckets_to_feed_the_hamsters/solution.rs
impl Solution {
    const HAMSTER: u8 = b'H';
    const EMPTY: u8 = b'.';

    pub fn minimum_buckets(hamsters: String) -> i32 {
        let bytes = hamsters.as_bytes();
        let len = bytes.len();
        let mut i: usize = 0;
        let mut cnt: usize = 0;

        while i < len {
            if bytes[i] == Self::HAMSTER {
                if i + 1 < len && bytes[i + 1] == Self::EMPTY {
                    i += 3;
                    cnt += 1;
                } else if i > 0 && bytes[i - 1] == Self::EMPTY {
                    cnt += 1;
                    i += 1;
                } else {
                    return -1;
                }
            } else {
                i += 1;
            }
        }

        cnt.try_into().unwrap()
    }
}
